internal/utils/meta.go

```diff
@@ -31,3 +35,35 @@ func AddAnnotation(obj metav1.Object, key string, value string) bool {
 	}
 	return false
 }
+
+// IsOwnedBy returns true if the object has an owner ref for the provided owner
+func IsOwnedBy(obj, owner metav1.Object) bool {
+	ownerRefs := obj.GetOwnerReferences()
+	for i := range ownerRefs {
+		if owner.GetUID() == ownerRefs[i].UID {
+			return true
+		}
+	}
+	return false
+}
+
+// ToggleOwnerReference adds or remove an owner reference for the given owner based on the first argument.
+// The function return true if the owner reference list had changed and false it it didn't
+func ToggleOwnerReference(on bool, obj, owner metav1.Object, scheme *runtime.Scheme) (bool, error) {
```

Comment:
is my understanding is correct? we are manipulating owners as not to remove the dependent if last owner is deleted?

Reply:
This is just a util that adds or removes from the owner list based on a provided flag

Reply:
Yes, I see atm this is being used for manipulating owners, so want to understand why do we need to explictly remove owner.

Reply:
For the cases when someone deleted the ClientProfile. We don't want to delete all owners and we don't want a "zombie" ownership that will prevent proper cleanup

Reply:
> We don't want to delete all owners
- there is no need to delete the owner, isn't it?

> we don't want a "zombie" ownership that will prevent proper cleanup
- when all the owners are deleted then only the dependent is removed by k8s and the lib (ctrlutil) afaik takes care of updating the owner w/ same name but different uid